src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+/// <summary>
+/// A struct to simplify holding all of the information
+/// about a package reference when using list
+/// </summary>
+public struct PRPackage
+{
+    public string package;
+    public VersionRange requestedVer;
+    public string requestedVerStr;
+    public NuGetVersion resolvedVer;
+    public NuGetVersion suggestedVer;
+    public bool deprecated;
+    public bool autoRef;
```

Comment:
I'm fully convinced why do you even need this class, most of these details are already there as part of LibraryDependency... May be going further in the code could help me understand that

Reply:
Deprecated, suggestedVersion, the string to requested version are not part of LibraryDependency, so I'm not sure if there is a better way to group all of the info together